doofinder/includes/class-update-on-save.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+namespace Doofinder\WP;
+
+use Doofinder\WP\Update_On_Save_Index;
+use Doofinder\WP\Log;
+use Doofinder\WP\Post;
+use Doofinder\WP\Multilanguage\Multilanguage;
+use Doofinder\WP\Settings\Accessors;
+
+class Update_On_Save {
+
+    /**
+    * Creates the database table for storing update on save information if it doesn't exist.
+    */
+    public static function create_update_on_save_db() {
+        global $wpdb;
+        $table_name = $wpdb->prefix . 'doofinder_update_on_save';
+        
+        $table_exists = $wpdb->get_var("SHOW TABLES LIKE '" . $table_name . "'");
+
+        if ($table_exists === NULL) {
+            // The table does not exist, we create it
+            $charset_collate = $wpdb->get_charset_collate();
+            $sql = "CREATE TABLE $table_name (
+                post_id INT NOT NULL,
+                type_post VARCHAR(255),
+                type_action VARCHAR(255),
+                PRIMARY KEY (post_id)
+            ) $charset_collate;";
+            require_once(ABSPATH . 'wp-admin/includes/upgrade.php');
+            dbDelta($sql);
+        }
+
+        update_option( 'doofinder_update_on_save_last_exec', date('Y-m-d H:i:s') );
+	}
+
+    /**
+     * Register hooks and actions.
+     */
+    public static function register_hooks() {
+        add_action('wp_insert_post', function ($post_id, \WP_Post $post, $updated) {
+            self::add_item_to_update($post_id, $post, $updated);
+        }, 99, 3);
+    }
+
+    /**
+     * Determines if the current post should be indexed and adds it to the update queue.
+     *
+     * @param int $post_id The ID of the post.
+     * @param WP_Post $post The post object.
+     * @param bool $updated Whether the post has been updated.
+     */
+	public static function add_item_to_update($post_id, $post, $updated) {
+
+        $log = new Log( 'update_on_save.txt' );
+        $update_on_save_index = new Update_On_Save_Index();
+
+        if (Settings::is_update_on_save_enabled()) {
+            $doofinder_post = new Post($post);
+            $log->log( 'Add this item to update on save: ' . print_r($doofinder_post, true)); 
+
+            self::add_item_to_db($doofinder_post, $post_id, $post->post_status, $post->post_type);
+
+            if(self::allow_process_items()) {
+                $log->log( 'We can send the data. '); 
+                $update_on_save_index->launch_doofinder_update_on_save();
+                return self::clean_update_on_save_db();
+            }
+        }
+
+        return;
```

Comment:
This returns something or nothing. Is it right? And why are you doing an empty return ?

Reply:
You're right, we really don't need to give anything back